primitives/header-chain/tests/implementation_match.rs

```diff
@@ -267,14 +270,15 @@ fn same_result_when_authority_equivocates_twice_in_a_round() {
 		Ok(()),
 	);
 	// original implementation returns non-empty GHOST
-	assert_matches!(
+	// https://github.com/paritytech/finality-grandpa/commit/36e409bcb76b41344be5b284732b529cd087432b#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759L496-R521
+	assert_eq!(
 		finality_grandpa::validate_commit(
 			&justification.commit,
 			&full_voter_set(),
 			&AncestryChain::new(&justification.votes_ancestries),
 		)
-		.map(|result| result.ghost().cloned()),
-		Ok(Some(_))
+		.map(|result| result.is_valid()),
+		Ok(false)
```

Comment:
I think there is `Ok(true)`

Reply:
the is_valid() is more strict than checking ghost() is Some(_). assert will fail if true.